hydra/grammar/OverrideLexer.g4

```diff
@@ -43,29 +44,28 @@ BRACE_CLOSE: WS? '}';
 VALUE_COLON: WS? ':' WS? -> type(COLON);
 VALUE_EQUAL: WS? '=' WS? -> type(EQUAL);
 
-// Types
-fragment NZ_DIGIT: [1-9];
-fragment INT_PART: DIGIT+;
-fragment FRACTION: '.' DIGIT+;
-fragment POINT_FLOAT: INT_PART? FRACTION | INT_PART '.';
-fragment EXPONENT: [eE] [+-]? DIGIT+;
-fragment EXPONENT_FLOAT: ( INT_PART | POINT_FLOAT) EXPONENT;
-FLOAT: [-]?(POINT_FLOAT | EXPONENT_FLOAT | [Ii][Nn][Ff] | [Nn][Aa][Nn]);
-INT: [-]? ('0' | (NZ_DIGIT DIGIT*));
+// Numbers.
+
+fragment POINT_FLOAT: INT_UNSIGNED '.' | INT_UNSIGNED? '.' DIGIT (('_')? DIGIT)*;
+fragment EXPONENT_FLOAT: (INT_UNSIGNED | POINT_FLOAT) [eE] [+-]? INT_UNSIGNED;
+FLOAT: [+-]? (POINT_FLOAT | EXPONENT_FLOAT | [Ii][Nn][Ff] | [Nn][Aa][Nn]);
+INT: [+-]? INT_UNSIGNED;
```

Comment:
These changes are related to what was discussed over chat with @omry (w.r.t. underscores & the + sign). I also condensed the definitions a bit to have fewer lines, as personally I actually find it easier to see at a glance what each type of float looks like, but I can revert back to more fragments if needed.